apps/sync/test/unit/adapters/bookmarks_test.js

```diff
@@ -284,69 +299,68 @@ suite('sync/adapters/bookmarks >', () => {
     });
   });
 
-  test('update - Add two records with the same URL and delete one', done => {
+  test('update - Add three records with the same URL and delete one', done => {
     var bookmarksAdapter = DataAdapters.bookmarks, store;
-    var deletedQueue = ['UNIQUE_ID_2'];
     Promise.resolve().then(() => {
-      var i = 1;
-      testCollectionData.unshift({
-        id: 'UNIQUE_ID_' + i,
-        last_modified: 100 + i * 10,
-        payload: {
-          id: 'UNIQUE_ID_' + i,
-          type: 'bookmark',
-          bmkUri: 'http://example.com/',
-          title: 'Example ' + i + ' Title'
-        }
-      });
-      i++;
-      testCollectionData.unshift({
-        id: 'UNIQUE_ID_' + i,
-        last_modified: 100 + i * 10,
-        payload: {
-          id: 'UNIQUE_ID_' + i,
-          type: 'bookmark',
-          bmkUri: 'http://example.com/',
-          title: 'Example ' + i + ' Title'
-        }
-      });
-      i++;
-      testCollectionData.unshift({
-        id: 'UNIQUE_ID_' + i,
-        last_modified: 100 + i * 10,
-        payload: {
+      for (var i = 1; i <= 3; i++) {
+        testCollectionData.unshift({
           id: 'UNIQUE_ID_' + i,
-          type: 'bookmark',
-          bmkUri: 'http://example.com/',
-          title: 'Example ' + i + ' Title'
-        }
-      });
-      return bookmarksAdapter.update(kintoCollection);
+          last_modified: 100 + i * 10,
+          payload: {
+            id: 'UNIQUE_ID_' + i,
+            type: 'bookmark',
+            bmkUri: 'http://example.com/',
+            title: 'Example ' + i + ' Title'
+          }
+        });
+      }
+      return bookmarksAdapter.update(kintoCollection,
+          { readonly: true, userid: 'foo' });
     }).then(result => {
       assert.equal(result, false);
       var mTime = testCollectionData[0].last_modified;
-      assert.equal(asyncStorage.mItems[BOOKMARKS_COLLECTION_MTIME], mTime);
+      assert.equal(asyncStorage.mItems['foo' + BOOKMARKS_COLLECTION_MTIME],
+          mTime);
       assert.equal(updateBookmarksSpy.callCount, 1);
       return Promise.resolve();
     }).then(() => {
-      var latestModified = testCollectionData[0].last_modified + 10000;
-      var deletedRecords = [];
-      deletedQueue.forEach((synctoId, i) => {
-        deletedRecords.push({
-          id: synctoId,
-          last_modified: latestModified + 10000 * i,
+      testCollectionData = [
+        {
+          id: 'UNIQUE_ID_2',
+          last_modified: testCollectionData[0].last_modified + 10000,
           payload: {
             deleted: true,
-            id: synctoId
+            id: 'UNIQUE_ID_2'
           }
-        });
-      });
-      testCollectionData = deletedRecords.concat(testCollectionData);
```

Comment:
This fixture was incorrect, it was `[ newRecord2, record3, record2, record1 ]` instead of `[ newRecord2, record3, record1 ]`.